### ISSUE TITLE
[CRASH] Issue with revamped phantoms

### ISSUE THREAD

**Issue by Jukabofy** [lukebemish/revamped_phantoms#8]
**Describe the bug**
It is pretty unnoticeable, but it is happily a rare thing
**To Reproduce**
Install the mod and use it, it somehow crashes sometimes, looks it happen when it interact modded mobs or something related to the renderization
**Expected behavior**
is Impredictable

**Screenshots**

**Version:**
 - Forge
 - v0.2.3

**Additional context**

[crash-2022-05-09_17.06.13-server.txt](https://github.com/lukebemish/revamped_phantoms/files/8655722/crash-2022-05-09_17.06.13-server.txt)

**Comment by lukebemish** [comment]
This is definitely related to other mods, though it may be partially on my end. Can you do a binary search to try and find out which?

**Comment by Jukabofy** [comment]
I'm not really sure, this is weird, maybe is caused by rubidium or rubidium extras

**Comment by Jukabofy** [comment]
I'm not going to see the mod separately, Anyone else could open the issue again

**Comment by lukebemish** [comment]
I'll leave it open in case someone else happens to have the same issue

**Comment by sirben99** [comment]
Hey, i seem to have something similar with ProbablyChests mod. I had a crash where a phantom picked up a mimic and the server crashed. I have also made a issue on ProbablyChests issue tracker, just incase this is on theyr end.

I hope this is related to what was happening here. If not please let me know and ill make a new issue!

Revamped Phantoms version: 1.18.2-0.2.3
Probably Chests version: 1.18.2-0.5.1

Crash:
[crash-reports_crash-2022-07-21_20.26.08-server.txt](https://github.com/lukebemish/revamped_phantoms/files/9162650/crash-reports_crash-2022-07-21_20.26.08-server.txt)

**Comment by ArcanePigeon** [comment]
It seems to be having some kind of issue with converting my custom goals, not sure why. been trying everything to fix it for the past couple hours and nothing has worked.

**Comment by lukebemish** [comment]
Hmm, weird. I have basically stopped development on this mod at present; I will likely pick it up again at some point in the future. However, if someone happens to find a fix, feel free to let me know or PR something

**Comment by Feldor-Lex** [comment]
I've also encountered this issue, 1.18.2. I've got a copy of a world where this is happening, so if I have the free time to comb through the mods I can. Main mods which add creatures on my pack are Ice and Fire, Mowzie's Mobs, Mana and Artifice, Creeper Overhaul, if that helps. It could be that Revamped Phantoms bugs and crashes when carrying multiple different modded mobs, though.

**Comment by lukebemish** [comment]
Yeah, I'm pretty sure there's something that some modded entities are doing that I don't react right to, but I've been unable to track it down yet. I'll have to keep looking into it. Thanks for the information.

**Comment by lukebemish** [comment]
If anyone is able to reproduce this on the new 1.20.4 version let me know; I am hoping that the rewrite I did there fixed this but wouldn't bet on it.
